src/components/Sticky/sidebar.test.js

```diff
@@ -0,0 +1,248 @@
+  })
+
+  it('can apply positioning', () => {
+    let container = document.createElement('div')
+    let content = document.createElement('div')
+    let options = {}
+    let world = {
+      direction: 'DOWN',
+      lastScrollTop: 1,
+      tolerance: 0,
+      boundaries: {
+        containerTop: 0,
+        containerBottom: 2,
+        contentHeight: 0
+      }
+    }
+
+    content.classList.add('sidebar-scrolling-down')
+    applyPositioning(container, content, world, options)
+    expect(hasClass(content, 'sidebar-scrolling-down')).toBe(true)
+
+    content.classList.remove('sidebar-scrolling-down')
+    world.direction = 'UP'
+    applyPositioning(container, content, world, options)
+    expect(hasClass(content, 'sidebar-top-fixed')).toBe(true)
+
+    world.lastScrollTop = 0
+    world.boundaries.containerTop = 1
+    applyPositioning(container, content, world, options)
+    expect(hasClass(content, 'sidebar-top-fixed')).toBe(false)
+  })
+
+  // it('can scroll down with nothing', () => {
+  //   let container = document.createElement('div')
+  //   let content = document.createElement('div')
+  //   const world = {
+  //     tolerance: 100,
+  //     lastScrollTop: 0,
+  //     boundaries: {
+  //       containerTop: 0,
+  //       containerBottom: 0,
+  //       containerHeight: 0
+  //     }
+  //   }
+  //   scrollUp(container, content, world)
+  //   expect(hasClass(content, 'sidebar-top-fixed')).toBe(false)
+  //   expect(hasClass(content, 'sidebar-scrolling-up')).toBe(false)
+  // })
+
+  // it('can scroll down with after scrolling down', () => {
+  //   let container = document.createElement('div')
+  //   let content = document.createElement('div')
+  //   content.classList.add('sidebar-scrolling-down')
+  //   const world = {
+  //     tolerance: 100,
+  //     lastScrollTop: 0,
+  //     boundaries: {
+  //       containerTop: 0,
+  //       containerBottom: 0,
+  //       containerHeight: 0
+  //     }
+  //   }
+  //   scrollUp(container, content, world)
+  //   expect(hasClass(content, 'sidebar-top-fixed')).toBe(false)
+  //   expect(hasClass(content, 'sidebar-scrolling-down')).toBe(false)
+  //   expect(hasClass(content, 'sidebar-scrolling-up')).toBe(true)
+  // })
+
+  // it('can scroll down with after bottomed out', () => {
+  //   let container = document.createElement('div')
+  //   let content = document.createElement('div')
+  //   content.classList.add('sidebar-bottom-fixed')
+  //   const world = {
+  //     tolerance: 100,
+  //     lastScrollTop: 0,
+  //     boundaries: {
+  //       containerTop: 0,
+  //       containerBottom: 0,
+  //       containerHeight: 0
+  //     }
+  //   }
+  //   scrollUp(container, content, world)
+  //   expect(hasClass(content, 'sidebar-bottom-fixed')).toBe(false)
+  //   expect(hasClass(content, 'sidebar-top-fixed')).toBe(false)
+  //   expect(hasClass(content, 'sidebar-scrolling-down')).toBe(false)
+  //   expect(hasClass(content, 'sidebar-scrolling-up')).toBe(true)
+  // })
```

Comment:
Why are these scroll down tests commented out?  Seems like they should be included to finish out the test suite.  Also, just curious, but on lines 100, 118, and 137, should they be calling `scrollDown` instead of `scrollUp`?

Reply:
They probably shouldn't be. Raising from the dead with https://github.com/truetandem/e-QIP-prototype/pull/2215/commits/a9ac180879796e80278deafd518623d8dc9fd680